src/ut_control_plane.c

```diff
@@ -117,7 +117,19 @@ static cp_message_t* dequeue_message(ut_cp_instance_internal_t *pInternal)
     {
         pthread_cond_wait(&pInternal->queue_condition, &pInternal->queue_mutex);
     }
-    msg = &pInternal->message_queue[0];
+    //msg = &pInternal->message_queue[0];
+    msg = (cp_message_t*)malloc(sizeof(cp_message_t));
+    if (msg == NULL)
```

Comment:
put an assert in there, if you can't allocate memory the whole system isn't going to work.